mslib/mscolab/server.py

```diff
@@ -738,14 +738,16 @@ def reset_request():
 @APP.route("/metadata/<idp_identity_name>", methods=['GET'])
 def metadata(idp_identity_name):
     """Return the SAML metadata XML for the requested IDP"""
-    for idp_config in setup_saml2_backend.CONFIGURED_IDPS:
-        if idp_identity_name == idp_config['idp_identity_name']:
-            sp_config = idp_config['idp_data']['saml2client']
-            metadata_string = create_metadata_string(
-                None, sp_config.config, 4, None, None, None, None, None
-            ).decode("utf-8")
-            return Response(metadata_string, mimetype="text/xml")
-    return render_template('errors/404.html'), 404
+    if mscolab_settings.USE_SAML2:
```

Comment:
please use a decorator for all these if mscolab_settings.USE_SAML2 changes
see 5.3 of https://towardsdatascience.com/5-advanced-tips-on-python-decorators-113307d5a92c 
e.g. use a  verify_use_saml2

Reply:
I think this shouldn't be a decorator, but the if statement should be outside of the function definitions, so that if `mscolab_settings.USE_SAML2` is false the SAML specific routes are not defined at all. This would mean that only a single if statement is necessary and no explicit error handling is needed.

Reply:
Thank you.. Resolved